R/Model_wrappers.R

```
@@ -54,82 +128,1045 @@ run_model <- function(params = NULL, initial_states = NULL, time, intervention =
+        if('Human_MDA_nic' %in% intervention || 'Human_MDA_pzq' %in% intervention || 
+           'Human_test_and_treat' %in% intervention && !is.numeric(age_target_human_MDA) &&
+           !is.numeric(age_target_human_test_and_treat)) {
+          
+          if ('Human_MDA_nic' %in% intervention && !is.numeric(age_target_human_MDA) &&
+              !'Human_MDA_pzq' %in% intervention && !'Human_test_and_treat' %in% intervention) {
+            states <- intervention_event_state(states = tail_states, intervention = 'Human_MDA_nic', intervention_effect = int_effect_size_list)
+          }
+          
+          if ('Human_MDA_pzq' %in% intervention && !is.numeric(age_target_human_MDA) 
+              && !'Human_MDA_nic' %in% intervention && !'Human_test_and_treat' %in% intervention) {
+            states <- intervention_event_state(states = tail_states, intervention = 'Human_MDA_pzq', intervention_effect = int_effect_size_list)
+          }
+          
+          if ('Human_test_and_treat' %in% intervention && !is.numeric(age_target_human_MDA) &&
+              !'Human_MDA_nic' %in% intervention && !'Human_MDA_pzq' %in% intervention) {
+            states <- intervention_event_state(states = tail_states, intervention = 'Human_test_and_treat', intervention_effect = int_effect_size_list)
+          }
+          
+          # 2) Options with 2 human intervention selected
+          if('Human_MDA_nic' %in% intervention && !is.numeric(age_target_human_MDA) &&
+             'Human_test_and_treat' %in% intervention && !is.numeric(age_target_human_test_and_treat) &&
+             ! 'Human_MDA_pzq' %in% intervention) {
+            states <- intervention_event_state(states = tail_states, intervention, intervention_effect = int_effect_size_list)
+          }
+          
+          if ('Human_MDA_pzq' %in% intervention && !is.numeric(age_target_human_MDA) &&
+              'Human_test_and_treat' %in% intervention && !is.numeric(age_target_human_test_and_treat) &&
```

Can this whole lot be reduced to a single call of:
`states <- intervention_event_state(states = tail_states, intervention, intervention_effect = int_effect_size_list)`
?

---

Can you check this throughout this function. There is _a lot_ of logic! If we can get rid or simplify it will make the code much less error prone

---

Thanks, yes will simplify/cut this logic down as suggested. Have left extra logic in when want to throw a unique error message e.g., if user tries to specify both human MDA with praziquantel and niclosamide

---

Thanks, yes I will do this for the above example, and will try to reduce/remove the logic for the age-structured code (combining some of the suggestions above e.g., using any(), changing && to &. I have left some of the logic statements where we need a unique error message thrown (e.g., if user tries to specify both praziquantel and niclosamide in human_MDA() ).